dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/utils/ProcessUtils.java

```diff
@@ -38,368 +41,408 @@
+            String cmd = String.format("sudo kill -9 %s", getPidsStr(processId));
 
-      logger.info("process id:{}, cmd:{}", processId, cmd);
+            logger.info("process id:{}, cmd:{}", processId, cmd);
 
-      OSUtils.exeCmd(cmd);
+            OSUtils.exeCmd(cmd);
 
-      // find log and kill yarn job
-      killYarnJob(taskExecutionContext);
+            // find log and kill yarn job
+            killYarnJob(taskExecutionContext);
 
-    } catch (Exception e) {
-      logger.error("kill task failed", e);
-    }
-  }
-
-  /**
-   * get pids str.
-   *
-   * @param processId process id
-   * @return pids
-   * @throws Exception exception
-   */
-  public static String getPidsStr(int processId) throws Exception {
-    StringBuilder sb = new StringBuilder();
-    Matcher mat;
-    // pstree pid get sub pids
-    if (OSUtils.isMacOS()) {
-      String pids = OSUtils.exeCmd("pstree -sp " + processId);
-      mat = MACPATTERN.matcher(pids);
-    } else {
-      String pids = OSUtils.exeCmd("pstree -p " + processId);
-      mat = WINDOWSATTERN.matcher(pids);
+        } catch (Exception e) {
+            logger.error("kill task failed", e);
+        }
     }
 
-    while (mat.find()) {
-      sb.append(mat.group(1)).append(" ");
-    }
-    return sb.toString().trim();
-  }
-
-  /**
-   * find logs and kill yarn tasks.
-   *
-   * @param taskExecutionContext  taskExecutionContext
-   */
-  public static void killYarnJob(TaskExecutionContext taskExecutionContext) {
-    try {
-      Thread.sleep(Constants.SLEEP_TIME_MILLIS);
-      LogClientService logClient = null;
-      String log = null;
-      try {
-        logClient = new LogClientService();
-        log = logClient.viewLog(Host.of(taskExecutionContext.getHost()).getIp(),
-                Constants.RPC_PORT,
-                taskExecutionContext.getLogPath());
-      } finally {
-        if (logClient != null) {
-          logClient.close();
-        }
-      }
-      if (StringUtils.isNotEmpty(log)) {
-        List<String> appIds = LoggerUtils.getAppIds(log, logger);
-        String workerDir = taskExecutionContext.getExecutePath();
-        if (StringUtils.isEmpty(workerDir)) {
-          logger.error("task instance work dir is empty");
-          throw new RuntimeException("task instance work dir is empty");
+    /**
+     * get pids str.
+     *
+     * @param processId process id
+     * @return pids pid String
+     * @throws Exception exception
+     */
+    public static String getPidsStr(int processId) throws Exception {
+        StringBuilder sb = new StringBuilder();
+        Matcher mat = null;
+        // pstree pid get sub pids
+        if (OSUtils.isMacOS()) {
+            String pids = OSUtils.exeCmd("pstree -sp " + processId);
+            if (null != pids) {
+                mat = MACPATTERN.matcher(pids);
+            }
+        } else {
+            String pids = OSUtils.exeCmd("pstree -p " + processId);
```

Comment:
i think this "pstree " can be extracted to one variable,  then I can easily replace this variable.